src/Module/Api/Mastodon/Instance.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+namespace Friendica\Module\Api\Mastodon;
+
+use Friendica\Core\Config;
+use Friendica\Core\Protocol;
+use Friendica\Core\System;
+use Friendica\Database\DBA;
+use Friendica\Model\User;
+use Friendica\Module\Base\Api;
+use Friendica\Module\Register;
+use Friendica\Network\HTTPException;
+use Friendica\Util\Network;
+
+/**
+ * @see https://docs.joinmastodon.org/api/rest/instances/
+ */
+class Instance extends Api
+{
+	public static function init(array $parameters = [])
+	{
+		parent::init($parameters);
+	}
+
+	/**
+	 * @param array $parameters
+	 * @throws HTTPException\InternalServerErrorException
+	 */
+	public static function rawContent(array $parameters = [])
+	{
+		$app = self::getApp();
+
+                $register_policy = intval(Config::get('config', 'register_policy'));
+
+		$return = [
+			'uri' => $app->getBaseURL(),
```

Comment:
I currently don't understand this whole concept in all details.

Reply:
Mastodon provides a detailed list of the entities that the API provides, and we should follow the provided definitions. Creating a class with the according members is the best way to achieve this. Then you can have a static factory method to generate a Mastodon API entity object from our local data structures, so that there's only one spot this conversion occurs.

Just take example on `Api\Mastodon\Account`. From https://docs.joinmastodon.org/api/entities/#instance create an  `Api\Mastodon\Instance` class with the same attribute names, the correct type-hints, and then create in the same class a static method named `Instance::createFromNode(BaseURL $baseUrl. Config $config)` that does pretty much the same thing as you're doing in the module except it sets a new Instance object class members.

And you will return this object in the API module which will look pretty much like this (don't copy and paste):

```php
public static function rawContent(array $parameters = [])
{

  $baseUrl = self::getClass(BaseURL::class);
  $config = self::getClass(Config::class);
  $return = Instance::createFromNode($baseUrl, $config);

  System::jsonExit($return);
}
```